lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -4724,6 +4728,66 @@ FIRRTLType RefSubOp::inferReturnType(ValueRange operands,
       loc, "ref.sub op requires a RefType of vector or bundle base type");
 }
 
+FIRRTLType RWProbeOp::inferReturnType(ValueRange operands,
+                                      ArrayRef<NamedAttribute> attrs,
+                                      std::optional<Location> loc) {
+  auto typeAttr = getAttr<TypeAttr>(attrs, "type");
+  auto type = typeAttr.getValue();
+  auto baseType = type_dyn_cast<FIRRTLBaseType>(type);
+  if (!baseType)
+    return emitInferRetTypeError(loc, "must be base type, not ", type);
```

Comment:
Is the checking checking of `FIRRTLBaseType` necessary here given that the constraint on `type` as already being a `FIRRTLBaseType` in ODS?

Reply:
If this isn't necessary, then consider rolling in the uninferred width/uninferred reset checks below into the ODS description. It seems a little funky to be mixing argument constraints with return type inference. The one exception here is validation across arguments which don't make sense to capture with constraints.

Reply:
I'm honestly not sure (cc previous discussions/issues re:ODS-constraints/verifier/inferReturnTypes being executed in different orders sometimes?) so just wrote this a bit defensively.  Having ODS check this seems like a good idea regardless, though!

Reply:
I think as long as inference won't fail if the constraints are invalid, this should be okay to handle just in ODS I think. Will update accordingly soon!